app/atomic_svc.py

```diff
@@ -0,0 +1,234 @@
+        """
+        regex = re.compile(r'\$?PathToAtomicsFolder((?:/[^/ \n]+)+|(?:\\[^\\ \n]+)+)')
+        payloads = []
+        if regex.search(string_to_analyse):
+            fullpath, path = regex.search(string_to_analyse).group(0, 1)
+            path = self._normalize_path(path, platform)
+
+            # take path from index 1, as it starts with /
+            path = os.path.join(self.repo_dir, 'atomics', path[1:])
+
+            if os.path.isfile(path):
+                payload_name = self._handle_attachment(path)
+                payloads.append(payload_name)
+                string_to_analyse = string_to_analyse.replace(fullpath, payload_name)
+
+        return string_to_analyse, payloads
+
+    def _use_default_inputs(self, test, platform, string_to_analyse):
+        """
+        Look if variables are used in `string_to_analyse`, and if any variable was given
+        a default value, use it.
+        """
+        payloads = []
+        defaults = dict((key, val) for key, val in test.get('input_arguments', dict()).items())
+        while RE_VARIABLE.search(string_to_analyse):
+            full_var_str, varname = RE_VARIABLE.search(string_to_analyse).groups()
+            default_var = str(defaults.get(varname, dict()).get('default'))
+
+            if default_var:
+                default_var, new_payloads = self._catch_path_to_atomics_folder(default_var, platform)
+                payloads.extend(new_payloads)
+                string_to_analyse = string_to_analyse.replace(full_var_str, default_var)
+
+        return string_to_analyse, payloads
+
+    @staticmethod
+    def _handle_multiline_commands(cmd):
+        return cmd.replace('\n', ';')
+
+    async def _prepare_cmd(self, entries, test, platform, cmd):
+        """
+        Handle a command or a cleanup (both are formatted the same way), given in `cmd`.
+        Return the cmd formatted as needed and payloads we need to take into account.
+        """
+        payloads = []
+        cmd, new_payloads = self._use_default_inputs(test, platform, cmd)
+        payloads.extend(new_payloads)
+        cmd, new_payloads = self._catch_path_to_atomics_folder(cmd, platform)
+        payloads.extend(new_payloads)
+        cmd = self._handle_multiline_commands(cmd)
+        return cmd, payloads
+
+    async def _prepare_executor(self, entries, test, platform):
+        """
+        Prepare the command and cleanup, and return them with the needed payloads.
+        """
+        payloads = []
+
+        command, payloads_command = await self._prepare_cmd(entries, test, platform, test['executor']['command'])
+        cleanup, payloads_cleanup = await self._prepare_cmd(entries, test, platform, test['executor'].get('cleanup_command', ''))
+        payloads.extend(payloads_command)
+        payloads.extend(payloads_cleanup)
+
+        return command, cleanup, payloads
+
+    async def _save_ability(self, entries, test):
+        """
+        Return True iif an ability was saved.
```

Comment:
spelling on if

Reply:
that was actually meant for "if and only if" :)